File: SDKs/github.com/ghes-2.19/1.1.4/go/pkg/models/shared/repositorysubscription.go
package shared

import (
	"time"
)

// RepositorySubscription
// Repository invitations let you manage who you collaborate with.
type RepositorySubscription struct {
	CreatedAt     time.Time `json:"created_at"`
	Ignored       bool      `json:"ignored"`
	Reason        string    `json:"reason"`
	RepositoryURL string    `json:"repository_url"`
	Subscribed    bool      `json:"subscribed"`
	URL           string    `json:"url"`
}

File: SDKs/github.com/ghes-2.20/1.1.4/typescript/src/sdk/models/shared/repositorysubscription.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// RepositorySubscription
/** 
 * Repository invitations let you manage who you collaborate with.
**/
export class RepositorySubscription extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=created_at" })
  createdAt: Date;

  @SpeakeasyMetadata({ data: "json, name=ignored" })
  ignored: boolean;

  @SpeakeasyMetadata({ data: "json, name=reason" })
  reason: string;

  @SpeakeasyMetadata({ data: "json, name=repository_url" })
  repositoryUrl: string;

  @SpeakeasyMetadata({ data: "json, name=subscribed" })
  subscribed: boolean;

  @SpeakeasyMetadata({ data: "json, name=url" })
  url: string;
}
